NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
-                            # discard event if delta_C (angle off cherenkov cone) is too large
-                            if np.abs(delta_Cs[iS]) > self._cfg['speedup']['delta_C_cut']:
-                                logger.debug('delta_C too large, ray tracing solution unlikely to be observed, skipping event')
+                        # applies the detector response to the electric fields (the antennas are defined
+                        # in the json detector description file)
+                        apply_det_response_sim(sim_station, self._det, self._config, self.detector_simulation_filter_amp,
+                                            event_time=self._evt_time, time_logger=self.__time_logger,
+                                            detector_simulation_part1=self.detector_simulation_part1)
+                        logger.debug(f"adding sim_station to station {sid} for event group {event_group.get_run_number()}, channel {channel_id}")
+                        station.add_sim_station(sim_station)  # this will add the channels and efields to the existing sim_station object
+                        for evt in output_buffer[sid].values():
+                            trace_start_time = evt.get_station().get_channel(channel_ids[0]).get_trace_start_time()
+                            # determine the trigger that was used to determine the readout window
+                            trigger_name = None
+                            trigger_time = None
+                            pre_trigger_time = None
+                            for trigger in evt.get_station().get_triggers().values():
+                                if trigger.get_pre_trigger_times() is not None:
+                                    pre_trigger_time = trigger.get_pre_trigger_times()
+                                    trigger_time = trigger.get_trigger_time()
+                                    trigger_name = trigger.get_name()
+                                    break
+
+                            assert(trigger_time is not None)
+                            assert(trigger_name is not None)
+                            assert(trace_start_time == trigger_time)
+
+                            for sim_channel in sim_station.get_channels_by_channel_id(channel_id):
+                                tt = sim_channel.get_times()
+                                t0 = tt[0]
+                                t1 = tt[-1]
+                                pre_trigger_time = triggerTimeAdjuster.get_pre_trigger_times(sim_channel.get_id())
+                                t0_readout = trigger_time - pre_trigger_time
+                                t1_readout = t0_readout + self._det.get_number_of_samples(sid, sim_channel.get_id()) * self._det.get_sampling_frequency(sid, sim_channel.get_id())
+
+                                # determine if the two intervals have any overlap
+                                if max(t0, t0_readout) < min(t1, t1_readout):
```

Comment:
wont the triggerTimeAdjuster take care of this (later)

Reply:
resp. should we just add the sim_channels and run the tirggerTimeAdjuster?

Reply:
this is (much) more efficient. We have potentially many many event where we try to add new channels to. The timing will only overlap with one event (or no event at all).

Reply:
I do not understand what you mean. Maybe we can discuss during the call.